### PR TITLE
OSDOCS-8631: add support for iSCSI disks on OCI for AI

### PR DESCRIPTION
**Pull request by jldohmann**
Version(s): 4.15+

Issue: https://issues.redhat.com/browse/OSDOCS-8631

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information: dependent on #69901

### CODE REVIEW

**Comment by jldohmann** [comment]
@avishayt can you PTAL at this PR? it adds iSCSI disk support for OCI on AI for OCP 4.15. it's pretty sparse, i'm wondering if there's more that needs to be discussed here, hoping you can provide your input

cc: @dfitzmau

File: modules/enabling-isci-assisted-installer.adoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+[id="enabling-iscsi-assisted-installer_{context}"]
+= Enabling iSCSI for the root disk in {op-system}
+
+If you have an iSCSI image target you can use the Assisted Installer to install {product-title} directly on iSCSI disks.
+
+.Procedure
+
+. Boot a host with the discovery image. The Assisted Installer automatically connects to any iBFT iSCSI targets.
+
+. Select the iSCSI disks for installation. The Assisted Installer adds the kernel arguments `rd.iscsi.firmware=1` and `ip=ibft`.
```
Comment (avishayt) [inline modules/enabling-isci-assisted-installer.adoc:15]: > The Assisted Installer adds the kernel arguments `rd.iscsi.firmware=1` and `ip=ibft`.

I think this is too much detail.

File: modules/enabling-isci-assisted-installer.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_oci/installing-oci-assisted-installer.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="enabling-iscsi-assisted-installer_{context}"]
+= Enabling iSCSI for the root disk in {op-system}
```
Comment (avishayt) [inline modules/enabling-isci-assisted-installer.adoc:7]: What is {op-system}?  Is this section in the context of OCI?

Comment (jldohmann) [reply modules/enabling-isci-assisted-installer.adoc:7]: it's RHCOS

Comment (avishayt) [reply modules/enabling-isci-assisted-installer.adoc:7]: Is this section in the context of OCI?  I don't know how this module fits into the rest of the docs.

Comment (jldohmann) [reply modules/enabling-isci-assisted-installer.adoc:7]: @avishayt yes, this is OCI only since it's only included in the upcoming OCI docs

**Comment by adriengentil** [comment]
@jldohmann can we set this change on hold? we haven't finalized the installation process on iSCSI boot volumes yet using the Assisted-Installer, so some things may change in the meantime.

File: modules/enabling-isci-assisted-installer.adoc (hunk 2 above)
Comment (dfitzmau) [reply modules/enabling-isci-assisted-installer.adoc:7]: @faermanj : would you be OK to chime in on where this content could sit in our OCI documentation?

**Comment by jldohmann** [comment]
@adriengentil yeah for sure, no issues holding off on merging this: do you have a timeline for when development will finish? i'm wondering if this will make it in for 4.15 or not

cc: @ahardin-rh

**Comment by adriengentil** [comment]
> @adriengentil yeah for sure, no issues holding off on merging this: do you have a timeline for when development will finish? i'm wondering if this will make it in for 4.15 or not

We have some development left in assisted installer, and an issue is open on Oracle side. There is a risk that this feature it may slip after 4.15 is released. cc @makentenza

**Comment by dfitzmau** [comment]
> > @adriengentil yeah for sure, no issues holding off on merging this: do you have a timeline for when development will finish? i'm wondering if this will make it in for 4.15 or not
> 
> We have some development left in assisted installer, and an issue is open on Oracle side. There is a risk that this feature it may slip after 4.15 is released. cc @makentenza

Hi @adriengentil , sorry to chime in here but the AI and Agent-based Installer routes in VM environments is still good fro tech preview? Just the bare metal environment work could slip because of pending Oracle tasks?

**Comment by adriengentil** [comment]
@dfitzmau yes, there are no blockers for OCI + VMs, and it's planned to be tech-preview.
We still have on-going work with OCI + baremetal shapes requiring iSCSI boot volumes.

**Comment by jlebon** [comment]
> This is not limited to OCI from an RHCOS point of view.

Agreed. Ideally, this would be mentioned also in the manual installation docs (in which case we would indeed document the kargs to add, i.e. `rd.iscsi.firmware=1` for iBFT-enabled systems, or `rd.iscsi.initiator=... netroot=iscsi:...` for manual configuration -- these arguments are documented [in the dracut docs](https://man7.org/linux/man-pages/man7/dracut.cmdline.7.html)).

> What we are supporting specifically is the iscsi_bft driver, i.e. iSCSI devices which work with the iSCSI Boot Firmware Table driver.

Manual configuration should also work. Though we're currently only testing the iBFT flow. (Added that to https://github.com/coreos/fedora-coreos-tracker/issues/1651 -- /cc @jbtrystram).

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.16` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.15. And any PR going into `main` must also target the latest version branch (enterprise-4.16).

If the update in your PR does NOT apply to version 4.16 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main

**Comment by bergerhoffer** [comment]
The `branch/enterprise-4.18` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.17. And any PR going into `main` must also target the latest version branch (enterprise-4.18).

If the update in your PR does NOT apply to version 4.18 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Pull request closed without merging**